ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -86,44 +88,37 @@ class MySqlNamedBlobDb implements NamedBlobDb {
    * Select a record that matches a blob name (lookup by primary key).
    */
   private static final String GET_QUERY =
-      String.format("SELECT %s, %s, %s FROM %s WHERE %s", BLOB_ID, EXPIRES_TS, DELETED_TS, NAMED_BLOBS, PK_MATCH);
+      String.format("SELECT %s, %s, %s, %s FROM %s WHERE %s AND %s ORDER BY %s DESC LIMIT 1", BLOB_ID, VERSION,
+          EXPIRES_TS, DELETED_TS, NAMED_BLOBS, PK_MATCH, STATE_MATCH, VERSION);
 
   /**
    * Select records up to a specific limit where the blob name starts with a string prefix. The fourth parameter can
    * be used for pagination.
    */
   private static final String LIST_QUERY = String.format("SELECT %1$s, %2$s, %3$s, %4$s FROM %5$s "
-          + "WHERE (%6$s, %7$s) = (?, ?) AND %1$s LIKE ? AND %1$s >= ? ORDER BY %1$s ASC LIMIT ?", BLOB_NAME, BLOB_ID,
-      EXPIRES_TS, DELETED_TS, NAMED_BLOBS, ACCOUNT_ID, CONTAINER_ID);
+          + "WHERE (%6$s, %7$s) = (?, ?) AND %8$s AND %1$s LIKE ? AND %1$s >= ? ORDER BY %1$s ASC LIMIT ?", BLOB_NAME,
+      BLOB_ID, EXPIRES_TS, DELETED_TS, NAMED_BLOBS, ACCOUNT_ID, CONTAINER_ID, STATE_MATCH);
```

Comment:
This is getting harder to read and will get harder as we add more args. Is there a java equivalent of python format using named args ?  MessageFormat perhaps ?

Reply:
MessageFormat works to let you use curly brace replacements, but perhaps at this point it is more readable to just accept some duplication but write out the queries in full without any replacements.

Reply:
Yeah, and you could also say "select * from <tablename> where <colname1> ...". Basically select all columns and then filter out what you don't need the get() method below.

Reply:
If `select *` is used, please be careful about getting by column index in the `ResultSet`.  golden gate schemas have a few extra columns to keep in mind.

Reply:
Updated to only use the replacement for the 'from/where' clause (totally 5 replacements).